pkgs/development/compilers/dotnet/combine-packages.nix

```diff
@@ -14,6 +14,7 @@ dotnetPackages:
 let
   cli = builtins.head dotnetPackages;
   mkWrapper = callPackage ./wrapper.nix { };
+  executableName = "dotnet${stdenv.hostPlatform.extensions.executable}";
```

Comment:
This should override `meta.mainProgram` or use `meta.mainProgram` and change it on the source of it.